Mathlib/NumberTheory/ArithmeticFunction.lean

```diff
@@ -591,6 +595,37 @@ theorem pdiv_zeta [DivisionSemiring R] (f : ArithmeticFunction R) :
 
 end Pdiv
 
+section ProdToFinsetFactors
+
+/-- The map $n \mapsto \prod_{p \mid n} f(p)$ as an arithmetic function -/
+def prodToFinsetFactors [CommMonoidWithZero R] (f : ℕ → R) : ArithmeticFunction R where
+  toFun d := if d = 0 then 0 else ∏ p in d.factors.toFinset, f p
+  map_zero' := if_pos rfl
+
+open Std.ExtendedBinder
+
+/-- `∏ᵖ p ∣ n, f p` is custom notation for `prodToFinsetFactors f n` -/
+scoped syntax (name := bigproddvd) "∏ᵖ " extBinder " ∣ " term ", " term:67 : term
+scoped macro_rules (kind := bigproddvd)
+  | `(∏ᵖ $x:ident ∣ $n, $r) => `(prodToFinsetFactors (fun $x ↦ $r) $n)
+
+/-- `∏ᵖ f` is custom notation for `prodToFinsetFactors f` -/
```

Comment:
I'm not sure this notation is a good idea - it could also refer to the product over primes (rather than prime divisors of a fixed number)

Reply:
That's a fair concern. It was only used in one place so I've removed it. 

It does make me think about other possible notations for the bare arithmetic function. Something like `∏ᵖ p ∣ _, f p` or `∏ᵖ p ∣ ·, f p` comes to mind but they obviously interfere with much more fundamental notation.